src/response.rs

```diff
@@ -261,6 +290,22 @@ impl Response {
     /// # }
     /// ```
     pub fn into_reader(self) -> impl Read + Send {
+        let stream = self.take_stream();
+        self.make_reader(stream)
+    }
+
+    /// Replaces `Response::stream` with ResponseStream::Empty and returns the result.
+    fn take_stream(&self) -> ResponseStream {
+        let mut stream = self.stream.lock().unwrap();
+        assert!(
+            !matches!(*stream, ResponseStream::Empty),
+            "ResponseStream already taken"
+        );
+        std::mem::replace(&mut *stream, ResponseStream::Empty)
+    }
+
+    /// Constructs a reader given the headers in this Response and the provided ResponseStream.
+    fn make_reader(&self, stream: ResponseStream) -> Box<dyn Read + Send> {
```

Comment:
This could take owned `self`, to make clear it can only be called once per `Response`.

Reply:
It can't. I specifically made this variant of `make_reader` to _not_ take self so that we can create a reader without destroying the `Response` struct. See other comment.